components/ReplySearch/ReplySearch.js

```diff
@@ -0,0 +1,170 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import stringSimilarity from 'string-similarity';
+import RelatedReplies from '../RelatedReplies';
+import SearchArticleItem from './SearchArticleItem.js';
+
+import { tabMenuStyle } from '../../pages/article.styles';
+
+const SearchArticles = ({ onConnect, searchArticles }) => {
+  return (
+    <ul className="items">
+      {searchArticles.map(article => {
+        return (
+          <SearchArticleItem
+            key={article.get('id')}
+            article={article}
+            onConnect={onConnect}
+          />
+        );
+      })}
+      <style jsx>{`
+        .items {
+          list-style-type: none;
+          padding-left: 0;
+        }
+      `}</style>
+    </ul>
+  );
+};
+
+export default class ReplySearch extends PureComponent {
+  state = {
+    tab: false, // reply || article
+    search: '',
+  };
+
+  handleSearch = event => {
+    const { target: { value }, key } = event;
+
+    if (key === 'Enter') {
+      this.setState({ search: value });
+      this.props.onSearch(event);
+    }
+  };
+
+  handleTabChange = tab => () => {
+    this.setState({ tab });
+  };
+
+  renderTabMenu = () => {
+    const { articles, replies } = this.props;
+    const { tab } = this.state;
+    const replisCount = replies.size;
+    const articlesCount = articles.size;
+
+    return (
+      <ul className="tabs">
+        <li
+          onClick={this.handleTabChange('reply')}
+          className={`tab ${tab === 'reply' ? 'active' : ''} ${
+            replisCount === 0 ? 'disabled' : ''
+          }`}
+        >
+          {replisCount === 0 ? (
+            '查無相關文章'
+          ) : (
+            <span>
+              使用相關回應 <span className="badge">{replisCount}</span>
+            </span>
+          )}
+        </li>
+        <li
+          onClick={this.handleTabChange('article')}
+          className={`tab ${tab === 'article' ? 'active' : ''} ${
+            articlesCount === 0 ? 'disabled' : ''
+          }`}
+        >
+          {articlesCount === 0 ? (
+            '查無相關文章'
+          ) : (
+            <span>
+              瀏覽相關文章 <span className="badge">{articlesCount}</span>
+            </span>
+          )}
+        </li>
+        <li className="empty" />
+        <style jsx>{`
+          .tabs {
+            margin-top: 20px;
+          }
+        `}</style>
+        <style jsx>{tabMenuStyle}</style>
+      </ul>
+    );
+  };
+
+  renderSearchReplyTab = () => {
+    const { articles, replies, onConnect } = this.props;
+    const { tab } = this.state;
+
+    const getArticleSimilarity = relatedArticleText =>
+      stringSimilarity.compareTwoStrings(this.search, relatedArticleText);
```

Comment:
`this.search` is undefined. Isn't `search` in `this.state`?

Error screenshot:
<img width="1229" alt="2018-02-08 1 08 45" src="https://user-images.githubusercontent.com/108608/35930349-a250ca58-0c6c-11e8-9ff8-517b15190dde.png">

Reply:
HAHA... big reconstruct mistake, THX!!